.github/workflows/samples-rust.yaml

```diff
@@ -0,0 +1,40 @@
+name: Samples Rust
+
+on:
+  push:
+    paths:
+      - 'samples/client/petstore/rust/**'
+      - 'samples/server/petstore/rust-server/**'
+  pull_request:
+    paths:
+      - 'samples/client/petstore/rust/**'
+      - 'samples/server/petstore/rust-server/**'
+
+jobs:
+  build:
+    name: Build Rust
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        sample:
+          - samples/client/petstore/rust/hyper/petstore
+          - samples/client/petstore/rust/reqwest/petstore-async-middleware
+          - samples/client/petstore/rust/reqwest/petstore-async
+          - samples/client/petstore/rust/reqwest/petstore-awsv4signature
+          - samples/client/petstore/rust/reqwest/petstore
+          - samples/server/petstore/rust-server/output/multipart-v3
+          - samples/server/petstore/rust-server/output/no-example-v3
+          - samples/server/petstore/rust-server/output/openapi-v3
+          - samples/server/petstore/rust-server/output/ops-v3
+          - samples/server/petstore/rust-server/output/petstore-with-fake-endpoints-models-for-testing
+          - samples/server/petstore/rust-server/output/ping-bearer-auth
+          - samples/server/petstore/rust-server/output/rust-server-test
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
```

Comment:
We should consider setting an MSRV here, or ideally having a matrix of both MSRV and latest. Similar to:
https://github.com/actix/actix-web/blob/45b77c68195eb933231290a09e9f6a0cca56aad8/.github/workflows/ci.yml#L21-L23

We should only run clippy on the MSRV so we avoid the issues here:
https://github.com/OpenAPITools/openapi-generator/pull/13907

Reply:
thanks for sharing that. agreed with the matrix of both MSRV and the latest.

let's see if anyone has time to do it in a separate PR after merging this one.